java/arcs/core/policy/PolicyConstraints.kt

```diff
@@ -132,19 +132,22 @@ sealed class PolicyViolation(val policy: Policy, message: String) : Exception(
     "Policy ${policy.name} violated: $message"
 ) {
     /** Thrown when egress particles were found in the recipe that are not allowed by policy. */
-    class InvalidEgressParticle(
+    class InvalidEgressParticles(
         policy: Policy,
-        val particleNames: List<String>
+        val allowedEgresses: List<String>,
+        val invalidEgresses: List<String>
     ) : PolicyViolation(
         policy,
-        "Egress particle allowed by policy is ${policy.egressParticleName} but found: " +
-            particleNames.joinToString()
+        "Egress particles allowed by policy are " +
+        allowedEgresses.toSortedSet().joinToString(prefix = "{", postfix = "}") +
+        ", but found: " +
+        invalidEgresses.toSortedSet().joinToString(prefix = "{", postfix = "}")
```

Comment:
Minor nit: I'd flip the order here, to make the more relevant part of the error message (the invalid ones) appear first in the message (there could be lots of allowed ones, but only one invalid one)

Reply:
done.